### DISCUSSION TITLE
Have issues when uploading files to storage

### DISCUSSION THREAD

**Discussion by hallee9000** [supabase/supabase#2666] [Questions] [answered]
Hi there, I have two questions when uploading files. Can't figure them out :sob:

1. I can't upload file to a specific bucket. I have 4 public buckets and neither of them has a policy, but I can upload files to the other 3 buckets. Uploading to `specs` bucket will get `new row violates row-level security policy for table "objects"`

![image](https://user-images.githubusercontent.com/12868055/127741630-ebc0985b-bcd5-4624-9d35-d4e79d0e9ce8.png)
![image](https://user-images.githubusercontent.com/12868055/127741648-9e6e5f24-e277-47cb-bade-79c0ded22143.png)

2. I got `application/json` as an image's content type after uploading a file from blob. Core code is below.

```js
// imageData is Uint8Array
const blob = new Blob([imageData], {
    type: 'image/png'
  })

  const fileExt = fileName.split('.').pop()
  const uniqueName = `${uuid()}.${fileExt}`
  const { data, error } = await supabase.storage
    .from('heron')
      .upload(uniqueName, blob, {
        contentType: 'image/png'
      })
```

![image](https://user-images.githubusercontent.com/12868055/127741623-7fbfbaa8-8008-4e42-a3dc-1dc25d7f88b0.png)

**Comment by dshukertjr** [comment, marked as answer]
Hi @leadream !

Hmm, it is odd that you were able to upload to 3 buckets, but not to `spec`. It seems like the error message is suggesting that there is something going on with the `objects` row level security. Could you maybe share your security policy on `objects`? You can find it below the policies of your individual buckets.

**Reply by EfosaSO** [reply]
I've just encountered this row-level security issue but on my policies everything Is blank and no longer showing my created policies.

**Reply by hallee9000** [reply]
@dshukertjr Also blank here. I remember I've created some policies for the other 3 buckets but can't see them after recent dashboard updates.

![image](https://user-images.githubusercontent.com/12868055/127757009-1190cc8d-9e94-40b6-8eae-f3b3028aae48.png)

**Reply by dshukertjr** [reply]
Hey @leadream and @EfosaSO !

Thanks for reporting this problem. I have opened an issue [here](https://github.com/supabase/supabase/issues/2670) so hopefully this gets fixed soon!

In the meanwhile, you will need security policy to allow insert on `storage.objects` table in order to have your users upload images to your buckets. This goes for any of the public buckets as well. Public only means that anyone can view, but doesn't mean anyone can upload. You can get started with any of the sample security policies found when you press "New policy" button. 

<img width="1364" alt="Screen Shot 2021-08-01 at 12 09 40" src="https://user-images.githubusercontent.com/18113850/127757849-c134df43-ae02-4865-9942-8c05f1891c44.png">

Although that doesn't really make sense why @leadream  you were able to upload to 3 buckets but not to `specs` bucket. Maybe once the issue on the policy editor gets fixed and we are able to view the security policy, we can investigate more!

**Reply by hallee9000** [reply]
@dshukertjr Thanks for looking into. I've created a new policy for bucket `specs` and it worked. For the image's content type issue I found it's a mistake from my code, I used the auth headers for `storage-js`.

**Reply by EfosaSO** [reply]
@dshukertjr I've created that policy but still get the row-level security issue and I still can't see the policy on my dashboard.

This is my error `message: "new row violates row-level security policy (USING expression) for table \"objects\""`
